### PR TITLE
Sync develop changes to hdf5_1_14

### PR DESCRIPTION
**Pull request by lrknox**

### CODE REVIEW

File: .github/workflows/hdfeos5.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: hdfeos5 dev
```
Comment (byrnHDF) [inline .github/workflows/hdfeos5.yml:1]: dev -> 1.14

Comment (byrnHDF) [inline .github/workflows/hdfeos5.yml]: dev -> 1.14

File: .github/workflows/hdfeos5.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+# Triggers the workflow on push or pull request or on demand
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
+    branches: [ develop ]
```
Comment (byrnHDF) [inline .github/workflows/hdfeos5.yml]: develop -> 1_14

File: release_docs/RELEASE.txt [hunk 3]
```diff
@@ -98,7 +98,15 @@ New Features
 
     Documentation:
     --------------
-    -
+    - Documented that leaving HDF5 threads running at termination is unsafe
```
Comment (byrnHDF) [inline release_docs/RELEASE.txt:101]: Did this not get into 1.14.5?

File: .github/workflows/hdfeos5.yml (hunk 1 above)
Comment (lrknox) [inline .github/workflows/hdfeos5.yml]: name: hdfeos5 1.14
```suggestion
name: hdfeos5 dev
```

Comment (lrknox) [inline .github/workflows/hdfeos5.yml]:
```suggestion
name: hdfeos5 1.14
```

File: .github/workflows/hdfeos5.yml (hunk 2 above)
Comment (lrknox) [inline .github/workflows/hdfeos5.yml]:
```suggestion
    branches: [ hdf5_1_14 ]
```

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**